_books/ion-1-1/src/modules.md

```diff
@@ -1,5 +1,121 @@
+
+#### Syntax
+```ion
+(macro_table arg1 arg2 ... argN)
+```
+#### Processing
+
+When the `macro_table` clause is encountered, the reader constructs an empty list. The arguments to the clause are then processed from left to right.
+
+For each `arg`:
+* **If the `arg` is a `macro` clause**, the clause is processed and the resulting macro definition is appended to the end of the macro table being constructed.
+* **If the `arg` is the name of a module**, the macro definitions in that module's macro table are appended to the end of the macro table being constructed.
+* **If the `arg` is anything else**, the reader must raise an error.
+
+
+Macro definitions being added to the macro table must have a unique name. If a macro is added whose name conflicts with one already present in the table, the reader must raise an error.
+
+### `macro`
+
+The `macro` clause defines a new macro.
+
+#### Syntax
+```ion
+(macro name signature template)
+```
+
+For more information, see _[Defining macros](macros/defining_macros.md)_.
+
+<!-- TODO: `import` -->
+<!-- TODO: `module` -->
+<!-- TODO: `export` -->
+
+## Grammar
+
+Literals appear in `code blocks`. Terminals are described in _italic text_.
+
+| Production           |     | Body                                                   |
+|----------------------|-----|--------------------------------------------------------|
+| module               | ::= | `(module ` module-name module-body `)`                 |
+| module-body          | ::= | import* module* symtab? mactab?                        |
+| import               | ::= | `(import` module-name catalog-name catalog-version `)` |
+| module               | ::= | `(module ` module-name `)`                             |
+| symtab               | ::= | `(symbol_table ` symtab-item* `)`                      |
+| symtab-item          | ::= | module-name \| symbol-def-seq                          |
+| symbol-def-seq       | ::= | _a list of unannotated text values (string/symbol)_    |
+| mactab               | ::= | `(macro_table ` mactab-item* `)`                       |
+| mactab-item          | ::= | module-name \| macro-def \| macro-export               |
+| macro-def            | ::= | `(macro ` macro-name signature tdl-template `)`        |
+| macro-export         | ::= | `(export ` macro-ref macro-name? `)`                   |
+| catalog-name         | ::= | _unannotated string_                                   |
+| catalog-version      | ::= | _unannotated int_                                      |
+| module-name          | ::= | _unannotated idenfitier symbol_                        |
+| macro-ref            | ::= | macro-name \| qualified-macro-name                     |
+| macro-name           | ::= | _unannotated idenfitier symbol_                        |
+| qualified-macro-name | ::= | _annotated idenfitier symbol_                          |
```

Comment:
May as well make this more precise, since we can easily do so.
```suggestion
| qualified-macro-name | ::= | module-name `::` macro-name                            |
```